problem1.py

```diff
@@ -0,0 +1,16 @@
+class Solution:
+        
+    def powerset(self, nums, value, result):
+        if nums:
+            for idx, i in enumerate(nums):
+                value.append(i)
+                result.append(value[:])
+                self.powerset(nums[idx+1:], value, result)
+                value.pop()
+        
+    def subsets(self, nums: List[int]) -> List[List[int]]:
+        result = [[]]
+        
+        self.powerset(nums,[], result)
+        return result
```

Comment:
Time complexity: O(2^N)
Space Complexity : O(2^N) where N is the number of digits
Algorithm: For every choice add the choice to a list and add the list to a result. After that call backtrack function again for the next n-1 elements

Reply:
Problem Faced: I had to add a copy of the value variable (line 7). Since it is passed through values through reference by default.